File: captureService/setting.py
import pymongo

class Setting:
    areas = [[0, 128, 0, 120]]
    threshold = [20]
    amount = [10]
    minCount = [10]
    countOn = [0]
    code = ""
    codeUsed = False
    prev = None
    imgCount = 0
    heldFrames = []
    fps = 0
    name = "hello"
    connection = ""
    active = False
    buffer = 0
    bufferUse = False
    buffered = []
    blur = 21
    boxJump = 150
    debug = True
    bufferBefore = 25
    bufferAfter = 15
    noMoveRefreshCount = 5

setting = Setting()

def update():
    x = mycol.find_one()
    print(x)

    setting.areas = x["area"]
    setting.threshold = x["threshold"]
    setting.amount = x["amount"]
    setting.minCount = x["mincount"]
    setting.fps = x["fps"]
    setting.name = x["name"]
    setting.connection = x["connection"]
    setting.active = x["motion"]

    setting.blur = x["blur"]
    setting.boxJump = x["boxJump"]
    setting.debug = x["debug"]
    setting.bufferBefore = x["bufferBefore"]
    setting.bufferAfter = x["bufferAfter"]
    setting.noMoveRefreshCount = x["noMoveRefreshCount"]


def connect():
    global mycol
    myclient = pymongo.MongoClient("mongodb://localhost:27017/")
    mydb = myclient["camera"]
    mycol = mydb["settings"]

File: captureService/setting.go
package main

import (
	"context"
	"time"

	"go.mongodb.org/mongo-driver/bson"
)

type settings struct {
	Name               string
	Connection         string
	FPS                int
	MinCount           int
	Motion             bool
	Blur               int
	Debug              bool
	BufferBefore       int
	BufferAfter        int
	NoMoveRefreshCount int
	Zones              []zone
}
type zone struct {
	X1          int
	Y1          int
	X2          int
	Y2          int
	Threshold   int
	BoxJump     int
	SmallIgnore int
	Area        int
}

func getSetting() settings {
	conn := databaseClient.Database("doorbell")
	db := conn.Collection("setting")
	filter := bson.M{"_id": 0}
	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
	doc := db.FindOne(ctx, filter)
	cancel()
	var s settings
	doc.Decode(&s)
	return s
}

func genTestSetting() settings {
	z := zone{10, 10, 500, 500, 20, 400, 2, 50}
	zo := make([]zone, 1)
	zo[0] = z
	s := settings{"test", "", 5, 3, true, 21, true, 5, 5, 3, zo}
	return s
}
